src/json/json.cc

```diff
@@ -624,6 +731,56 @@ int Command_JsonSet(ValkeyModuleCtx *ctx, ValkeyModuleString **argv, int argc) {
     return ValkeyModule_ReplyWithSimpleString(ctx, "OK");
 }
 
+int Command_JsonMSet(ValkeyModuleCtx *ctx, ValkeyModuleString **argv, int argc) {
+    ValkeyModule_AutoMemory(ctx);
+
+    MSetCmdArgsList args;
+    JsonUtilCode rc = parseAndValidateMSetCmdArgs(ctx, argv, argc, &args);
+    if (rc != JSONUTIL_SUCCESS) {
+        if (rc == JSONUTIL_WRONG_NUM_ARGS)
+            return ValkeyModule_WrongArity(ctx);
+        else
+            return ValkeyModule_ReplyWithError(ctx, jsonutil_code_to_message(rc));
+    }
+
+    // Apply changes
+    for (size_t i = 0; i < args.num_keys; i++) {
+
+        if (args.args_list[i].is_root_path){  // root doc
+            // parse incoming JSON string
+            JDocument *doc;
+            rc = dom_parse(ctx, args.args_list[i].json, args.args_list[i].json_len, &doc);
```

Comment:
Here we are calling dom_parse() second time. There are two problems:
1. not efficient: parsing is expense. If the json is large, there's a performance hit.
2. it makes doc size tracking inaccurate. We track the doc size earlier inside parseAndValidateMSetCmdArgs and stored the value there. Here we just use that value. However, technically "this dom_parse()" is not same as "that dom_parse()).

So, as suggested earlier, we should call dom_parse only once and store the `*doc` in args.

Reply:
second thought. We can delay on performance enhancement. It's okay for now to call dom_parse() twice. However, we must track the memory usage here instead of inside parseAndValidateMSetCmdArgs.